### DISCUSSION TITLE
Error part question during installation

### DISCUSSION THREAD

**Discussion by gnzmzn321** [cloudforet-io/community#126] [Q&A] [answered]
Hello
I have a question during installation
Following the quick install guide on cloudforet.io
I'm asking because there's something different from the guide in the "kubectl logs initializer-5f5b7b5cdc-lnjkl-n spaceone" part.

When looking up a pod showing the token value with the logs command, an error message is displayed as shown below.
A workaround is required.
Thank you.
![화면 캡처 2023-06-01 142422](https://github.com/cloudforet-io/cloudforet/assets/104878970/1db4dccd-ea6e-4327-b1e6-6171ac512b46)

**Comment by choonho** [comment]
Hello.

My guess is that you executed initializer (helm install initializer cloudforet/spaceone-initializer -n spaceone -f initializer.yaml) before all PODs was ready.

In the guide,
You have to check every POD is "RUNNING" status, except scheduler PODs.

![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/268af5d8-01b0-4324-b49d-6051e8290ce4)

If you want to re-initialize,

1) Delete Database POD (mongodb-xxxx), "kubectl delete pod mongodb-XXXXXXXXXXX"
2) After checking every POD is "RUNNING" status except scheduler PODs.
3) execute initalizer "helm install initializer cloudforet/spaceone-initializer -n spaceone -f initializer.yaml"

If you have a problem, I will check the guide again.

Thank you.

**Comment by gnzmzn321** [comment]
I don't know if the problem is Mongo in Helm, but the status doesn't change to running
I checked the information about Mongod with description



please tell me what's wrong ..
-----원본 메시지-----
보낸 사람: ***@***.***>
받는 사람: ***@***.***>;
Cc: ***@***.***>; ***@***.***>;
보낸 날짜: 2023-06-02 (금) 14:42:44 (GMT+09:00)
제목: Re: [cloudforet-io/cloudforet] 설치 중 오류 부분 질문 (토론 #126)

안녕하세요.
모든 POD가 준비되기 전에 초기화 프로그램(helm install initializer cloudforet/spaceone-initializer -n spaceone -f initializer.yaml)을 실행한 것 같습니다.
가이드에서
스케줄러 POD를 제외한 모든 POD가 "RUNNING" 상태인지 확인해야 합니다.
재초기화를 원하시면,

데이터베이스 POD 삭제(mongodb-xxxx), "kubectl delete pod mongodb-XXXXXXXXXXX"
스케줄러 POD를 제외한 모든 POD가 "RUNNING" 상태인지 확인한 후.
이니셜라이저 실행 "helm install initializer cloudforet/spaceone-initializer -n spaceone -f initializer.yaml"

문제가 있으면 안내를 다시 확인하겠습니다.
감사합니다.
—
이 이메일에 직접 회신하거나 Git
![화면 캡처 2023-06-05 175846](https://github.com/cloudforet-io/cloudforet/assets/104878970/c618ad77-96d9-4616-9094-1224e95de854)
Hub에서 보거나 구독을 취소 하십시오 .
스레드를 작성했기 때문에 이 메시지를 받고 있습니다.메시지 ID: <cloudforet-io/cloudforet/repo-discussions/126/comments/6067866 @ github . com>
![화면 캡처 2023-06-05 175950](https://github.com/cloudforet-io/cloudforet/assets/104878970/77b08664-e0f8-4c39-b5e4-987c0c121123)

**Comment by choonho** [comment]
# POD status

Can you list your POD status?

Command is ***kubectl get pod -n spaceone***

# Kubernetes version

What is your minikube version?

~~~
minikube version
~~~


# Could you follow up installation guide again?
After cleaning up all minikube.

~~~
minikube delete --all
~~~

https://cloudforet.io/docs/setup_operation/quick_install/

**Comment by gnzmzn321** [comment]
The installation is complete as follows.
But unlike the guide, I only have administration and I don't see other projects or anything like that. Is this a problem with installation or what is it is?

![ad](https://github.com/cloudforet-io/cloudforet/assets/104878970/adcde0c6-468e-47d5-aec1-091df849a0c0)

**Comment by choonho** [comment]
Cloudforet supports multi-tenancy like slack.
We call this as "domain".

In the minikube installation, we create two tenants, "root" as default and "spaceone" domain.
This multi-tenancy configuration is in "values.yaml"

* DOMAIN_NAME
* DOMAIN_NAME_REF

~~~
console:
  production_json:
    # If you don't have a service domain, you refer to the following 'No Domain & IP Access' example.
    CONSOLE_API:
      ENDPOINT: http://{Change Your EIP}:8081
    CONSOLE_API_V2:
      ENDPOINT: http://{Change Your EIP}:8082
    DOMAIN_NAME: spaceone
    DOMAIN_NAME_REF: spaceone
~~~

This screen looks like "root" domain which is system domain for system management.
In a minikube installation, we automatically forward web console to "spaceone" domain, but you were forwared to "root" domain.


There are two possible reason,

# Failure of initialization (Step 5 in a guide doc)

* Please check the log of initializer-xxxx POD
* If there is EROOR in the log, please attach the log
* If there is no ERROR in the log, the problem is error of Step 6.

# Failure of upgrade (Step 6 in a guide doc)

* Wrong vaules.yaml content for upgrade. (forwarding to ROOT domain is default configuration)
* restarting of console POD is not executed. 

Please restart helm upgrade again.
 
~~~
helm upgrade cloudforet cloudforet/spaceone -n spaceone -f values.yaml
kubectl delete po -n spaceone -l app.kubernetes.io/instance=cloudforet
~~~

**Comment by choonho** [comment]
One more important thing,
The minikube installation guide is for developer only, which means the database is not persistent.

If there is a restart of mongodb POD, every data in the database will be deleted.

**Reply by gnzmzn321** [reply]
I didn't install it on minikube, I proceeded in a cloud environment!

I want to know why only Administration is checked in the environment
![ds](https://github.com/cloudforet-io/cloudforet/assets/104878970/235571b5-f6e7-463c-a1f2-31b8a0254e2e)

**Comment by choonho** [comment]
If you didn't follow up minikube installation. You need your own DNS domain like example.com
To use multi-tenancy, we provides this as {tenant name}.example.com

To access "spaceone" tenant, the access URL is "spaceone.example.com".

To use this, you need your own ingress controller and additional ingress configuration and different ***values.yaml***

This is hard to explain without your values.yaml configuration.

See https://github.com/cloudforet-io/charts

**Comment by choonho** [comment, marked as answer]
I confused.
The administration menu is correct one, if you logged in "Sign in for Root Account".

![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/25d857d3-245e-4a63-b1fe-510cb846b9f0)

![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/cbaed1f9-e3a2-423b-b2d8-3ecf6d62568d)

After creating the new user, log-out "admin", then log-in new user,

![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/11dd42b1-0b57-4566-a34c-5b27133b99f5)


![image](https://github.com/cloudforet-io/cloudforet/assets/3993088/67fe21e9-7b99-4412-b9a3-f82b1addaa3c)

**Comment by nimitseth** [comment]
Need some help here please : Am getting below Error (use GKE for installation)
![e1](https://github.com/cloudforet-io/cloudforet/assets/47523831/af1a40a5-1fec-47fb-8e33-8d42e9ddb9b1)

**Reply by nimitseth** [reply]
in values.yaml what endpoint should i give, if i don't have a external DNS name ?

    CONSOLE_API:
      ENDPOINT: https://console.api.example.com       # Change the endpoint
    CONSOLE_API_V2:
      ENDPOINT: https://console-v2.api.example.com    # Change the endpoint

**Reply by choonho** [reply]
@nimitseth If you use Kubernetes like EKS, AKE, or GKE, Domain name for cloudforet is mandatory. Otherwise browser cannot access to endpoint.

if you can create an application load balancer, you can use a Domain name of the load balancer.

What is the IP of screenshot? If the IP is a load balancer, you can use like

~~~
CONSOLE_API:
   ENDPOINT: http://a.b.c.d
CONSOLE_API_V2:
   ENDPOINT: http://x.y.z.f
~~~

**Reply by nimitseth** [reply]
@choonho : Really appreciate your quick response. I don't have any real domain name as we are testing it.
Below is the services details, we have for all these pods. 

Can i use the LB name (highlight in yellow) as endpoint 

![e4](https://github.com/cloudforet-io/cloudforet/assets/47523831/caece287-dc0b-467d-8b20-2cc0357c6d30)


![e2](https://github.com/cloudforet-io/cloudforet/assets/47523831/0cec72fb-b067-4a57-b3ab-88f4092393f1)

**Reply by choonho** [reply]
@nimitseth Yes, your load balancer has external & public IP address.

Use like 
~~~
CONSOLE_API:
   ENDPOINT: http://34.31.226.8
CONSOLE_API_V2:
   ENDPOINT: http://34.80.210.97
~~~

> http instead of https

**Reply by nimitseth** [reply]
@choonho : Okay, I have made the changes as below in values.yaml as below and re-ran 
helm upgrade cloudforet cloudforet/spaceone -n spaceone -f values.yaml

![e5](https://github.com/cloudforet-io/cloudforet/assets/47523831/9cd19eb3-812c-4bbc-93ed-d4a68fea3146)
Still getting this error  :(

![image](https://github.com/cloudforet-io/cloudforet/assets/47523831/981b3479-8014-400a-807e-a2ebca727bc2)
![image](https://github.com/cloudforet-io/cloudforet/assets/47523831/edd58960-86a7-424e-a43f-3a77e50599c0)
![image](https://github.com/cloudforet-io/cloudforet/assets/47523831/16ed5d14-0276-460d-807e-8e775df3db7b)

Do i need to run "kubectl delete po -n spaceone -l app.kubernetes.io/instance=cloudforet" as well ?

**Reply by choonho** [reply]
Yes. may be
